### ISSUE TITLE
Login with new client left me in a broken state

### ISSUE THREAD

**Issue by grempe** [keybase/keybase-issues#1902]
I saw that the new Go client had come out so I tried to give it a go on my laptop which is the only device I have keybase installed on.  I removed the old binaries and installed the new via homebrew and did a 'keybase login'.

This started the wizard for setting things up and I chose the option to sign with my GPG keys.  This is where things went south.  I use a Yubikey to store my gpg sub-keys.  Unfortunately there is a bug in OS X where the card daemon sometimes gets in a bad state and needs to be restarted.  It was this way yesterday when I was doing the keybase setup.  So when key base prompted for my gpg keys and tried to use them it failed and this left my install in a bad state.

I was kind of logged in to keybase, but no device had been registered and no nacl keys were generated.

When I tried to login again it seemed to work, I could interact with key base, but trying to list devices failed with an error since there was no device created through the API I guess.

I ended up trying to reset myself by removing all key base data in my home dir and in the Application Support dir and removing and re-installing keybase.

Ultimately this worked but it was touch and go.

I think the request for what needs to be fixed is that the login and setup operations for the client need to be atomic. Either all steps pass in initial setup and are applied, or you fail them all and nothing gets done.

**Comment by maxtaco** [comment]
Which version did you experience these problems on?

**Comment by grempe** [comment]
$ keybase --version
keybase version 1.0.5-0

Installed from homebrew.

**Comment by maxtaco** [comment]
Ok thanks. We actually just made the change to atomic provisioning for the
reasons you mentioned but apparently there are still issues.  Thanks for
your feedback.

**Comment by xbglowx** [comment]
Are the documented steps on how to fix this. I believe I am seeing the same problem.

```
keybase device add
What type of device would you like to connect this computer with?

(1) Desktop or laptop
(2) Mobile phone

Choose a device type: 1

Enter the verification code from your other device here.  To get
a verification code, run 'keybase login' on your other device.

Verification code: ▶ ERROR No device found No device found for device ID 85a7333cec7414b05a026ffe637c0618
```

**Comment by maxtaco** [comment]
Blasting away your config should work.  See `keybase config info` for where it is; then remove it; then run `keybase ctl restart` to restart the daemon, and then try again.

Though of course we'd love to know why this happened.  What version are you running?  Have you recently upgraded?  Thanks so much for your help.

**Comment by maxtaco** [comment]
Also, what series of commands did you use to get to this state?  Did any previous commands fail with visible warnings? Thanks!

**Comment by xbglowx** [comment]
```
keybase version
Client:  1.0.5-0
Service: 1.0.5-0
```

I recently upgraded today and I was in the process of adding the device when it asked me to login first. I tried to login, but forgot my credentials, so it failed. Not sure if that was actually the problem, but my scrollback buffer is not big enough for me to examine.

**Comment by maxtaco** [comment]
@xbglowx digging a bit deeper, it seems like you've failed to successfully provision any devices, since your profile shows 0 device keys active.  So likely when you tried to `login` on the machine you've shown above, login failed.  Did it leave any error messages?  What version is running there?

BTW, the only way you can provision is via gpg, since you don't host your private key with our syncing service.

One last question: are you upgrading from another Go version, or directly from the old Node client?

Thanks so much for your patience and help here. Sorry for any problems.

**Comment by xbglowx** [comment]
Correct, I had to do gpg login and this time it worked but the previous time it didn't. I believe it failed, since I didn't have the correct keybase passphrase. I just tried it again after deleting they keybase config and it worked.

Previously, I had keybase installed via brew. Not sure if this was a Go version or not.

No worries about the problems. You have been very helpful. If you want me to try to reproduce this problem, I can try to reset and fail the login again, but not sure if that will help if my device (laptop) is now registered.
